### PR TITLE
Refactoring Deploy Cluster to use the a generic Wizard

### PR DESCRIPTION
**Pull request by aasimkhan30**
Most of the wizard specific logic is moved to deploy Cluster model. This will eliminate a lot of circular dependency that was present in the Tools and Eula Wizard PR #12951.

The wizard looks the same as it is the same code.
image
image
image

### CODE REVIEW

File: extensions/resource-deployment/src/ui/deployClusterWizard/deployClusterWizardModel.ts [hunk 1]
```diff
@@ -2,19 +2,93 @@
...
+	initialize(): void {
+		this.wizard.setPages(this.getPages());
+		this.wizard.wizardObject.generateScriptButton.hidden = true;
+		this.wizard.wizardObject.doneButton.label = localize('deployCluster.ScriptToNotebook', "Script to Notebook");
+	}
+	async onOk(): Promise<void> {
+		await this.scriptToNotebook();
+	}
+	onCancel(): void {
+		throw new Error('Method not implemented.');
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/ui/deployClusterWizard/deployClusterWizardModel.ts:79]: Is this what we really want?

File: extensions/resource-deployment/src/ui/resourceTypeWizard.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+		public azdataService: IAzdataService,
+		public notebookService: INotebookService,
+		public toolsService: IToolsService,
+		public platformService: IPlatformService) {
+		this.wizardObject = azdata.window.createWizard(resourceType.displayName, resourceType.name, 'wide');
+		this.model = this.getResourceProviderModel();
+	}
+
+
+	public getResourceProviderModel(): ResourceTypeModel {
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/ui/resourceTypeWizard.ts:52]: If this is returning undefined then the return type should be | undefined

File: extensions/resource-deployment/src/ui/resourceTypeWizard.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+			try {
+				disposable.dispose();
+			}
+			catch (error) {
+				errorOccurred = true;
+				console.error(error);
+			}
+		});
+
+		if (errorOccurred) {
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/ui/resourceTypeWizard.ts:126]: Is there a reason you're adding this? Generally if we're closing the wizard I'm not sure we need to display anything to user if an error happens (and in general I wouldn't expect errors to happen normally so having this check is a little odd)

File: extensions/resource-deployment/src/ui/resourceTypeWizard.ts [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+		this.toDispose.push(disposable);
+	}
+
+}
+
+
+
+export class ResourceTypePage extends WizardPageBase<ResourceTypeWizard>{
+	public initialize(): void {
+		throw new Error('Method not implemented.');
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/ui/resourceTypeWizard.ts:141]: This class should probably be abstract instead and then not have a definition for initialize - having this throw isn't really going to ever be a valid scenario I think

**Review by Charles-Gagnon — COMMENTED** [review]
Please include screenshots of the new wizard as well

File: extensions/resource-deployment/src/ui/resourceTypeWizard.ts (hunk 2 above)
Comment (aasimkhan30) [reply extensions/resource-deployment/src/ui/resourceTypeWizard.ts:52]: For now, yes. Later on when all providers are migrated, it won't return undefined

File: extensions/resource-deployment/src/ui/resourceTypeWizard.ts (hunk 3 above)
Comment (aasimkhan30) [reply extensions/resource-deployment/src/ui/resourceTypeWizard.ts:126]: I got this code from the current Wizard Base. What you are suggesting makes sense. I will remove it now.

File: extensions/resource-deployment/src/ui/resourceTypeWizard.ts (hunk 2 above)
Comment (Charles-Gagnon) [reply extensions/resource-deployment/src/ui/resourceTypeWizard.ts:49]: You should still make sure the return types match what can actually be returned - even if it's temporary. You can remove it later on if needed.

**Review by Charles-Gagnon — APPROVED** [review]

**Review by alanrenmsft — APPROVED** [review]

**Pull request merged**